src/main/java/com/aws/greengrass/device/session/SessionFactory.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.device.session;
+
+import com.aws.greengrass.device.exception.AuthenticationException;
+
+import java.util.Map;
+
+public interface SessionFactory {
+    Session createSession(Map<String, String> credentialMap) throws AuthenticationException;
```

Comment:
you want to enforce string->string? I think you may want to be a bit more generic, String->Object, or even make it generic String->T

Reply:
Didn't want to overthink this one too much to start. It should be easy enough to change later

Reply:
It is our common base. It includes several extensions including a log exception protector and extra thread protector. Just good to have as best practice everywhere, even in unit tests which don't use GG stuff.